### PR TITLE
feat: add patreon ga

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/Donation/patreon-button.tsx [hunk 1]
```diff
@@ -10,7 +10,13 @@ const { patreonClientId }: { patreonClientId: string | null } = envData as {
   patreonClientId: string | null;
 };
 
-const PatreonButton = (): JSX.Element | null => {
+interface PatreonButtonPropTypes {
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/patreon-button.tsx:13]:
```suggestion
interface PatreonButtonProps {
```

File: client/src/components/Donation/patreon-button.tsx [hunk 2]
```diff
@@ -10,7 +10,13 @@ const { patreonClientId }: { patreonClientId: string | null } = envData as {
...
+  postPatreonRedirect: () => void;
+}
+
+const PatreonButton = ({
+  postPatreonRedirect
+}: PatreonButtonPropTypes): JSX.Element | null => {
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/patreon-button.tsx:19]:
```suggestion
}: PatreonButtonProps): JSX.Element | null => {
```

File: client/src/components/Donation/patreon-button.tsx [hunk 3]
```diff
@@ -30,6 +36,7 @@ const PatreonButton = (): JSX.Element | null => {
       className='patreon-button link-button'
       data-patreon-widget-type='become-patron-button'
       href={href}
+      onClick={() => postPatreonRedirect()}
```
Comment (ojeytonwilliams) [inline client/src/components/Donation/patreon-button.tsx:39]:
```suggestion
      onClick={postPatreonRedirect}
```

**Review by ojeytonwilliams — COMMENTED** [review]
Just some nitpicks:

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

File: client/src/components/Donation/patreon-button.tsx [hunk 4]
```diff
@@ -30,6 +36,7 @@ const PatreonButton = (): JSX.Element | null => {
       className='patreon-button link-button'
       data-patreon-widget-type='become-patron-button'
       href={href}
+      onClick={postPatreonRedirect}
```
Comment (ShaunSHamilton) [inline client/src/components/Donation/patreon-button.tsx:39]: @ahmadabdolsaheb If you are sure about giving an `a` element an `onclick`, then I am fine with it, but I would advise this be a `button` with a side-effect redirecting to `href`.

Comment (ojeytonwilliams) [reply client/src/components/Donation/patreon-button.tsx:39]: Isn't that an a11y no-no, though, making an `a` into a `button`?  This really is just a link that's dressed up to look like a button.

Comment (ShaunSHamilton) [reply client/src/components/Donation/patreon-button.tsx:39]: > Isn't that an a11y no-no, though, making an a into a button?

Yes, but it goes both ways. This is not just a hyperlink for navigation, it invokes a `postPatreonRedirect` side-effect.

Comment (ojeytonwilliams) [reply client/src/components/Donation/patreon-button.tsx:39]: True enough.  It's kind of a weird hybrid, but it seems to be primarily a link (I click on it and my browser heads of to Patreon).

I don't know enough a11y stuff to be sure, though.

Comment (ahmadabdolsaheb) [reply client/src/components/Donation/patreon-button.tsx:39]: I think it should be fine.

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 

P.S. _There is a typo in `DonateForm` Line `216` of 'submition'_ - I do not know if this affects the analytics in any way?

**Comment by ahmadabdolsaheb** [comment]
There is some historical data associated with that action.
We will address this when moving to GA4 to Q1 2022 :)

**Pull request merged**